### PR TITLE
Login state and check, landing page

### PR DESCRIPTION
**Pull request by danbigthink**

### CODE REVIEW

File: api/app/config/environment.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const DEVELOPMENT_ENV = 'development';
```
Comment (seeker25) [inline api/app/config/environment.js:1]: You're probably going to want to get docker and manage.sh to work with this 

https://github.com/bcgov/court-of-appeal/blob/b631797066ae05fd395c85f83527b79533a1a5e1/docker/manage#L227

So it's straight forward for us to put the environment variables in Openshift.

**Review by seeker25 — CHANGES_REQUESTED** [review]

**Comment by seeker25** [comment]
So far so good a few concerns:

image

From the picture: 

1.	Extra spacing needs to be fixed
2.	A few requests being sent? Too many?
3.	Spacing at the bottom? 



4.	I don’t see the logout button anymore?
5.	Also I think there is a way to pass a ?next= parameter when logging in? That way it can take us directly back to the dashboard instead of going back to the main page, having to call AJAX is_logged_in and then redirecting?

**Comment by seeker25** [comment]
I've fixed a few.. I see the logout doesn't show up  because i'm missing the secrets

**Comment by seeker25** [comment]
It looks like landing.css is bleeding over other styles.

**Comment by danbigthink** [comment]
> I've fixed a few.. I see the logout doesn't show up because i'm missing the secrets

- I have no idea what the BCEID_LOGIN_URL env value should be. Can you update that or send me the url?

- I can take out the in memory service, should we leave it there if NODE_ENV === 'test'?

- The changes look good, thanks. You had mentioned you didn't want the logout link showing in the header (once logged in) so I hid it unless NODE_ENV===development. 

- I had tried the redirect to /dashboard the way you have it and that didn't work, using start-dev with craco.config.js causes all typed in url's into the address bar to return a 404 if you try to use external nav (ie. http://localhost:3000/court-of-appeal can be typed into the address bar but http://localhost:3000/court-of-appeal/dashboard cannot). 

- I'll get this all running in docker.

**Comment by seeker25** [comment]
I've fixed the second to last point.. just doing one small tweak for the landing page spacing right now

**Comment by seeker25** [comment]
I don't think we use the BCEID_LOGIN URL?

**Comment by seeker25** [comment]
"You had mentioned you didn't want the logout link showing in the header (once logged in) so I hid it unless NODE_ENV===development."

Sorry it shouldn't show up if on the landing page, should probably still show up for users though because they'll need to log out.

I think most of the outstanding issues have been fixed, just get it working under docker (with environment variables).. and we'll merge it.

**Comment by danbigthink** [comment]
@seeker25 
- Changing that now - Sorry it shouldn't show up if on the landing page, should probably still show up for users though because they'll need to log out.

**Comment by seeker25** [comment]
Excellent, looks pretty good man!

**Comment by danbigthink** [comment]
> Excellent, looks pretty good man!

Thanks, I'll finish up the docker and build stuff tomorrow afternoon.

File: openshift/templates/api/api-deploy.test.param [hunk 2]
```diff
@@ -19,7 +19,7 @@ TAG_NAME=test
 # KEYCLOAK_CLIENT_ID=
 # KEYCLOAK_SECRET=
 # COOKIE_SECRET=
-# SM_LOGOUT_URL_PREFIX=
+SM_LOGOUT_URL_PREFIX=https://logontest.gov.bc.ca/clp-cgi/logoff.cgi
```
Comment (seeker25) [inline openshift/templates/api/api-deploy.test.param:22]: Probably need this done for DEV / PROD as well?

**Comment by seeker25** [comment]
Url for PROD is https://logon.gov.bc.ca/clp-cgi/logoff.cgi

File: openshift/templates/api/api-deploy.overrides.sh [hunk 3]
```diff
@@ -22,6 +22,7 @@ if createOperation; then
   readParameter "KEYCLOAK_SECRET - Please provide the KeyCloak Client Secret.  The default is a blank string." KEYCLOAK_SECRET "" "false"
 
   readParameter "COOKIE_SECRET - Please provide the Cookie Secret for the environemnt.  The default is a blank string." COOKIE_SECRET "" "false"
+  readParameter "SM_LOGOUT_URL_PREFIX - Please provide the Siteminder logout url prefix for the environemnt.  The default is a blank string." SM_LOGOUT_URL_PREFIX "" "false"
```
Comment (seeker25) [inline openshift/templates/api/api-deploy.overrides.sh]: Don't need this here if I remember correctly

File: openshift/templates/api/api-deploy.overrides.sh [hunk 4]
```diff
@@ -34,6 +35,7 @@ else
   writeParameter "KEYCLOAK_SECRET" "prompt_skipped" "false"
 
   writeParameter "COOKIE_SECRET" "prompt_skipped" "false"
+  writeParameter "SM_LOGOUT_URL_PREFIX" "prompt_skipped" "false"
```
Comment (seeker25) [inline openshift/templates/api/api-deploy.overrides.sh]: Don't need this here if I remember correctly

**Pull request merged**